frontend/src/App.tsx

```diff
@@ -1,33 +1,63 @@
-import React from "react";
-import { BrowserRouter as Router } from "react-router-dom";
-import styled, { createGlobalStyle } from "styled-components";
-import "./App.css";
-import PrivateRoutes from "./components/Routes/PrivateRoutes";
-import { mainBackgrund } from "./utils/Colors";
-import { ScrollToTop } from "./utils/scroll";
+import React, { createContext, useState, useEffect } from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
+import './App.css';
 
+import PrivateRoutes from './components/Routes/PrivateRoutes';
+import { mainBackgrund } from './utils/Colors';
+import { ScrollToTop } from './utils/scroll';
+import { light, dark } from './utils/Themes';
 const GlobalStyle = createGlobalStyle`
   body {
     background-color: ${mainBackgrund};
   }
 `;
+const GlobalStyleLight = createGlobalStyle`
+ body {
+    background-color:'white';
+  }
+`;
 
+const themesMap: any = { light, dark };
 const Container = styled.div`
-  font-family: "Roboto", sans-serif;
-  color: black;
+  font-family: 'Roboto', sans-serif;
+  color: ${(props) => props.theme.body};
   line-height: initial;
 `;
 
 const App: React.FC = () => {
+  const [currentTheme, setCurrentTheme] = useState('dark');
+  const theme = themesMap[currentTheme];
+
+  useEffect(() => {
+    const themeQuery = window.matchMedia('(prefers-color-scheme: light)');
+    setCurrentTheme(themeQuery.matches ? 'light' : 'dark');
+    themeQuery.addEventListener('change', ({ matches }) => {
+      setCurrentTheme(matches ? 'light' : 'dark');
+    });
+  }, []);
+
   return (
     <Router>
       <ScrollToTop />
-      <Container>
-        <GlobalStyle />
-        <PrivateRoutes />
-      </Container>
+      <ThemePreference.Provider value={{ currentTheme, setCurrentTheme }}>
```

Comment:
I rather seperate it to different file containing thr ThemeProvider

Reply:
I tried separating it and ran into many type errors and problems.